src/cmd/manage_nsfs.js

```diff
@@ -506,44 +464,31 @@ async function update_account(data, is_flag_iam_operate_on_root_account) {
+        await config_fs.delete_account_config_file(cur_name, data.access_keys);
     } else if (update_access_key) {
-        await native_fs_utils.update_config_file(fs_context, global_config.accounts_dir_path, cur_account_config_path, encrypted_data);
+        await config_fs.update_account_config_file(cur_name, parsed_data, parsed_data.access_keys, [cur_access_key]);
     }
-    // TODO: safe_unlink can be better but the current impl causing ELOOP - Too many levels of symbolic links
-    // need to find a better way for atomic unlinking of symbolic links
-    // handle atomicity for symlinks
-    await nb_native().fs.unlink(fs_context, cur_access_key_config_path);
-    await nb_native().fs.symlink(fs_context, new_account_relative_config_path, new_access_key_config_path);
     write_stdout_response(ManageCLIResponse.AccountUpdated, data);
 }
 
 async function delete_account(data) {
-    await manage_nsfs_validations.validate_account_args(global_config, data, ACTIONS.DELETE, undefined);
-
-    const fs_context = native_fs_utils.get_process_fs_context(global_config.config_root_backend);
-    const account_config_path = get_config_file_path(global_config.accounts_dir_path, data.name);
-    await native_fs_utils.delete_config_file(fs_context, global_config.accounts_dir_path, account_config_path);
-    if (has_access_keys(data.access_keys)) {
-        for (const access_key_object of data.access_keys) {
-            const access_key_config_path = get_symlink_config_file_path(global_config.access_keys_dir_path, access_key_object.access_key);
-            await nb_native().fs.unlink(fs_context, access_key_config_path);
-        }
-    }
-    write_stdout_response(ManageCLIResponse.AccountDeleted, '', {account: data.name});
+    await manage_nsfs_validations.validate_account_args(config_fs, data, ACTIONS.DELETE, undefined);
+    await config_fs.delete_account_config_file(data.name, data.access_keys);
+    write_stdout_response(ManageCLIResponse.AccountDeleted, '', { account: data.name });
 }
 
 async function get_account_status(data, show_secrets) {
-    await manage_nsfs_validations.validate_account_args(global_config, data, ACTIONS.STATUS, undefined);
+    await manage_nsfs_validations.validate_account_args(config_fs, data, ACTIONS.STATUS, undefined);
+    const options = { show_secrets, decrypt_secret_key: show_secrets };
+
     try {
-        const account_path = _.isUndefined(data.name) ?
-            get_symlink_config_file_path(global_config.access_keys_dir_path, data.access_keys[0].access_key) :
-            get_config_file_path(global_config.accounts_dir_path, data.name);
-        const config_data = await get_config_data(global_config.config_root_backend, account_path, show_secrets);
-        if (config_data.access_keys) config_data.access_keys = await nc_mkm.decrypt_access_keys(config_data);
+        const config_data = _.isUndefined(data.name) ?
```

Comment:
in general, there is no longer a need in js to use any function to check for undefined (not _.isUndefined and not util.isUndefined), and the best check is the trivial check - `data.name === undefined`. the need for functions was an old remain from back when undefined was not "well defined" in javascript but those times are long gone.

Reply:
Since it's out of scope for this PR and this PR blocks other tasks, I added it to a new NC short refactoring tasks github issue, feel free to add there whatever comes to your mind - https://github.com/noobaa/noobaa-core/issues/8264